pkg/services/auth/testdata/auth_token_mocks.go

```diff
@@ -0,0 +1,81 @@
+package testdata
```

Comment:
Unsure if this is a good idea:
* mock in its own package below the real implementation?
* in testdata package?
* must be available for both api and middleware packages

Reply:
Hm, usually I dont like this. It's a signal that our code structure/arch is a bit scattered. 

I'm not sure middlewares should be its own package. But we can discuss that separately. For this case, I think it's fine to expose a mock package but I think it should be named mocks. `testdata` is a reserved foldername for test data used in tests. Ex https://medium.com/@povilasve/go-advanced-tips-tricks-a872503ac859

Reply:
> It's a signal that our code structure/arch is a bit scattered. 

True that! :smile: 

Yeah middleware and api is a complete story on its own, but feels like we really need to dig into that sooner or later. Yes I know about testdata, but you've talked about that code in testdata doesn't get added to the compiled binary - may have misunderstood that.

Reply:
Moved this into testing.go in auth package according to suggestion in link you referenced.